drivers/gpio/gpio_eos_s3.c

```diff
@@ -0,0 +1,432 @@
+			cfg->xPadConf->ucPin = PAD_6;
+			cfg->xPadConf->ucFunc = PAD6_FUNC_SEL_GPIO_0;
+			break;
+		case GPIO_1:
+			cfg->xPadConf->ucPin = PAD_9;
+			cfg->xPadConf->ucFunc = PAD9_FUNC_SEL_GPIO_1;
+			break;
+		case GPIO_2:
+			cfg->xPadConf->ucPin = PAD_11;
+			cfg->xPadConf->ucFunc = PAD11_FUNC_SEL_GPIO_2;
+			break;
+		case GPIO_3:
+			cfg->xPadConf->ucPin = PAD_14;
+			cfg->xPadConf->ucFunc = PAD14_FUNC_SEL_GPIO_3;
+			break;
+		case GPIO_4:
+			cfg->xPadConf->ucPin = PAD_18;
+			cfg->xPadConf->ucFunc = PAD18_FUNC_SEL_GPIO_4;
+			break;
+		case GPIO_5:
+			cfg->xPadConf->ucPin = PAD_21;
+			cfg->xPadConf->ucFunc = PAD21_FUNC_SEL_GPIO_5;
+			break;
+		case GPIO_6:
+			cfg->xPadConf->ucPin = PAD_22;
+			cfg->xPadConf->ucFunc = PAD22_FUNC_SEL_GPIO_6;
+			break;
+		case GPIO_7:
+			cfg->xPadConf->ucPin = PAD_23;
+			cfg->xPadConf->ucFunc = PAD23_FUNC_SEL_GPIO_7;
+			break;
+		default:
+			return -EINVAL;
+		}
+	} else if (port_num == 1) {
+		switch (cfg->ucGpioNum) {
+		case GPIO_0:
+			cfg->xPadConf->ucPin = PAD_24;
+			cfg->xPadConf->ucFunc = PAD24_FUNC_SEL_GPIO_0;
+			break;
+		case GPIO_1:
+			cfg->xPadConf->ucPin = PAD_26;
+			cfg->xPadConf->ucFunc = PAD26_FUNC_SEL_GPIO_1;
+			break;
+		case GPIO_2:
+			cfg->xPadConf->ucPin = PAD_28;
+			cfg->xPadConf->ucFunc = PAD28_FUNC_SEL_GPIO_2;
+			break;
+		case GPIO_3:
+			cfg->xPadConf->ucPin = PAD_30;
+			cfg->xPadConf->ucFunc = PAD30_FUNC_SEL_GPIO_3;
+			break;
+		case GPIO_4:
+			cfg->xPadConf->ucPin = PAD_31;
+			cfg->xPadConf->ucFunc = PAD31_FUNC_SEL_GPIO_4;
+			break;
+		case GPIO_5:
+			cfg->xPadConf->ucPin = PAD_36;
+			cfg->xPadConf->ucFunc = PAD36_FUNC_SEL_GPIO_5;
+			break;
+		case GPIO_6:
+			cfg->xPadConf->ucPin = PAD_38;
+			cfg->xPadConf->ucFunc = PAD38_FUNC_SEL_GPIO_6;
+			break;
+		case GPIO_7:
+			cfg->xPadConf->ucPin = PAD_45;
+			cfg->xPadConf->ucFunc = PAD45_FUNC_SEL_GPIO_7;
+			break;
+		default:
+			return -EINVAL;
+		}
+	} else {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/*
+ * NOTE:
+ * Pin number defined in the device tree should be actually a GPIO number
+ * to avoid overflow in high level GPIO API.
+ * `gpio_num_extended` in range of 0 to 7 matches to `port_num == 0` PADs from
+ * `find_pad` function and `gpio_num_extended` in range of 8 to 15
+ * are associated with `port_num == 1` consequently.
+ */
+static int gpio_eos_s3_configure(const struct device *dev,
+				 gpio_pin_t gpio_num_extended,
+				 gpio_flags_t flags)
+{
+	ARG_UNUSED(dev);
+	int ret;
+	GPIOCfgTypeDef gpio_cfg;
+	PadConfig pad_conf;
+	int port_num = 0;
+
+	gpio_cfg.xPadConf = &pad_conf;
+
+	/* Check for an invalid gpio number */
+	if (gpio_num_extended >= MAX_GPIOS) {
+		return -EINVAL;
+	}
+
+	/* Determine "port" number */
+	if ((int)(gpio_num_extended / MAX_GPIOS_PER_PORT) == 1) {
+		port_num = 1;
+	}
+
+	/* Get actual GPIO number */
+	uint8_t gpio_num = gpio_num_extended % 8;
+
+	/* Check for an invalid pin configuration */
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
```

Comment:
`(flags & (GPIO_INPUT | GPIO_OUTPUT)) != 0` ?

Or `(flags & DIR_MASK) != 0` ?

Reply:
I would leave it as is. Checking against e.g. `(flags & DIR_MASK) != 0` will result in "some" value which will correspond to either `GPIO_INPUT` or `GPIO_OUTPUT`. The same check is performed in the mcux driver: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/gpio/gpio_mcux.c#L48

Reply:
OK, I get it: it doesn't support configuring in both directions.

This possibly should be:

    if (((flags & GPIO_INPUT) != 0) == ((flags & GPIO_OUTPUT) != 0))

because as-is if neither input nor output is specified it'll be configured as an output.

Alternatively just use the switch statement that mcux has after the `if` you point to (which is not necessary, since the switch statement that detects both unsupported configurations.)  Multiple drivers use the switch idiom.